### PR TITLE
Added simple attic watch store example

### PR DESCRIPTION
**Pull request by mannp**
This is a simple but working systemd service to upload the contents of the local buildbox-nix store to an attic cache, for use my other machines/nodes.

### CODE REVIEW

File: examples/attic-watch-store.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    serviceConfig = {
+      MemoryHigh = "5%";
+      MemoryMax = "10%";
+    };
+    script = ''
+      #!/run/current-system/sw/bin/bash
+      ATTIC_TOKEN=$(cat ${config.sops.secrets."attic/prod-auth-token".path})
+      ${pkgs.attic}/bin/attic login prod https://cache.<domain> $ATTIC_TOKEN
+      ${pkgs.attic}/bin/attic use prod
+      ${pkgs.attic}/bin/attic watch-store prod:prod
```
Comment (Mic92) [inline examples/attic-watch-store.nix:35]:
```suggestion
      exec ${pkgs.attic}/bin/attic watch-store prod:prod
```

File: examples/attic-watch-store.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  systemd.services.attic-watch-store = {
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network-online.target" ];
+    serviceConfig = {
+      MemoryHigh = "5%";
+      MemoryMax = "10%";
+    };
+    script = ''
+      #!/run/current-system/sw/bin/bash
```
Comment (Mic92) [inline examples/attic-watch-store.nix:31]:
```suggestion
      #!${pkgs.runtimeShell}
```

File: examples/attic-watch-store.nix (hunk 1 above)
Comment (Mic92) [inline examples/attic-watch-store.nix:35]:
```suggestion
      attic login prod https://cache.<domain> $ATTIC_TOKEN
      attic use prod
      exec attic watch-store prod:prod
```

File: examples/attic-watch-store.nix (hunk 2 above)
Comment (Mic92) [reply examples/attic-watch-store.nix:31]: nixos already inserts a shebang here.

File: examples/attic-watch-store.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    attic
+  ];
+
+  systemd.services.attic-watch-store = {
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network-online.target" ];
+    serviceConfig = {
+      MemoryHigh = "5%";
+      MemoryMax = "10%";
+    };
```
Comment (Mic92) [inline examples/attic-watch-store.nix:29]:
```suggestion
    };
    path = [ pkgs.attic ];
```

File: examples/attic-watch-store.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    script = ''
+      #!/run/current-system/sw/bin/bash
+      ATTIC_TOKEN=$(cat ${config.sops.secrets."attic/prod-auth-token".path})
```
Comment (Mic92) [inline examples/attic-watch-store.nix:32]:
```suggestion
      ATTIC_TOKEN=$(< $CREDENTIALS_DIRECTORY/prod-auth-token)
```

File: examples/attic-watch-store.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  environment.systemPackages = with pkgs; [
+    attic
+  ];
+
+  systemd.services.attic-watch-store = {
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network-online.target" ];
+    serviceConfig = {
+      MemoryHigh = "5%";
+      MemoryMax = "10%";
```
Comment (Mic92) [inline examples/attic-watch-store.nix:28]:
```suggestion
      MemoryMax = "10%";
      LoadCredential = "prod-auth-token:${config.sops.secrets."attic/prod-auth-token".path}";
```

File: examples/attic-watch-store.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  nix.settings.netrc-file = config.sops.secrets."attic/netrc-file-pull-push".path;
+
+  environment.systemPackages = with pkgs; [
+    attic
+  ];
+
+  systemd.services.attic-watch-store = {
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network-online.target" ];
+    serviceConfig = {
```
Comment (Mic92) [inline examples/attic-watch-store.nix:26]:
```suggestion
    serviceConfig = {
      DynamicUser = true;
```

**Comment by Mic92** [comment]
I added some suggestions above. I have not tested them yet. Could you try to apply them in your setup? Thanks!

**Comment by Mic92** [comment]
The CI error I can address later myself.

File: examples/attic-watch-store.nix [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  # Add netrc file for this machine to do its normal thing with the cache, as a machine.
+  nix.settings.netrc-file = config.sops.secrets."attic/netrc-file-pull-push".path;
+
+  environment.systemPackages = with pkgs; [
+    attic
+  ];
+
+  systemd.services.attic-watch-store = {
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network-online.target" ];
```
Comment (Mic92) [inline examples/attic-watch-store.nix:25]:
```suggestion
    after = [ "network-online.target" ];
    environment.HOME = "/var/lib/attic-watch-store";
```

File: examples/attic-watch-store.nix [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    serviceConfig = {
+      DynamicUser = true;
+      MemoryHigh = "5%";
+      MemoryMax = "10%";
+      LoadCredential = "prod-auth-token:${config.sops.secrets."attic/prod-auth-token".path}";
```
Comment (Mic92) [inline examples/attic-watch-store.nix:30]:
```suggestion
      LoadCredential = "prod-auth-token:${config.sops.secrets."attic/prod-auth-token".path}";
      StateDirectory = "attic-watch-store";
```

File: examples/attic-watch-store.nix (hunk 7 above)
Comment (Mic92) [reply examples/attic-watch-store.nix:25]: ^ This hopefully fixes the write error you are reporting.

**Comment by mannp** [comment]
Yes that works now for me 👍🏻

**Comment by Mic92** [comment]
Looks like attic-client is now in nixpkgs. So I just replaced the attic flake with that as it saves the user some rebuilds.

**Comment by Mic92** [comment]
@mergify queue

**Comment by mannp** [comment]
> Looks like attic-client is now in nixpkgs. So I just replaced the attic flake with that as it saves the user some rebuilds.

Okay with me, with the mental note that it may cause version mismatches if the flake module is used to configure the main attic service.

 #, inputs  is no longer required with the removal of the flake.

**Comment by Mic92** [comment]
The module also seems to use the server provided by nixpkgs from the looks of it: https://github.com/zhaofengli/attic/blob/717cc95983cdc357bc347d70be20ced21f935843/flake.nix#L183
So that should be reasonable in sync.

**Pull request merged**